// jquery-file-upload-middleware/lib/fileinfo.js
var fs = require('fs'),
    _ = require('lodash');

module.exports = function (options) {

    var FileInfo = function (file) {
        this.name = file.name;
        this.originalName = file.name;
        this.size = file.size;
        this.type = file.type;
        this.deleteType = 'DELETE';
    };

    FileInfo.prototype.validate = function () {
        if (options.minFileSize && options.minFileSize > this.size) {
            this.error = 'File is too small';
        } else if (options.maxFileSize && options.maxFileSize < this.size) {
            this.error = 'File is too big';
        } else if (!options.acceptFileTypes.test(this.name)) {
            this.error = 'Filetype not allowed';
        }
        return !this.error;
    };

    FileInfo.prototype.safeName = function () {

      /**
        var extname = require('path').extname(this.name)
        // Prevent directory traversal and creating hidden system files:
        this.name = Date.parse(new Date()) + Math.ceil(Math.random()*100) + extname;
        // Prevent overwriting existing files:
        while (fs.existsSync(options.baseDir() + '/' + this.name)) {
            this.name = '' + Date.parse(new Date()) + Math.ceil(Math.random()*100) + extname;
        }
        **/
    };

    FileInfo.prototype.setUrl = function(type, baseUrl) {
		var key = type ? type + 'Url' : 'url';
		this[key] = baseUrl + '/' + encodeURIComponent(this.name);
	}

	FileInfo.prototype.setPath = function(path) {
		this["path"] = path;
	}

	FileInfo.prototype.setUploadDir = function(uploadDir) {
		this["uploadDir"] = uploadDir;
	}
    return FileInfo;
};
